### PR TITLE
Deprecate helpers.hpp

### PR DESCRIPTION
**Pull request by chfast**
Depends on #357

### CODE REVIEW

File: include/evmc/helpers.hpp [hunk 1]
```diff
@@ -101,6 +109,7 @@ template <>
 struct hash<evmc_address>
 {
     /// Hash operator using FNV1a.
+    EVMC_DEPRECATED
```
Comment (gumb0) [inline include/evmc/helpers.hpp]: Can't it be applied to the entire `struct hash<evmc_address>`?

Comment (chfast) [reply include/evmc/helpers.hpp]: It can be.

File: test/unittests/test_helpers.cpp [hunk 2]
```diff
@@ -27,27 +26,6 @@ static constexpr size_t optionalDataSize =
...
 
-
-TEST(helpers, fnv1a)
-{
-    const uint8_t text[] = {'E', 'V', 'M', 'C'};
-    const auto h = fnv1a(text, sizeof(text));
-    EXPECT_EQ(h, sizeof(size_t) == 8 ? 0x15e05d6d22fed89a : 0xffaa6a9a);
-}
-
-TEST(helpers, is_zero)
```
Comment (axic) [inline test/unittests/test_helpers.cpp]: I don't think tests should be removed with deprecation. We still need to ensure nobody breaks the deprecated features.

Comment (axic) [reply test/unittests/test_helpers.cpp]: That being said, I think the test removed in https://github.com/ethereum/evmc/commit/279c1cd6e75b69947f33b33a95348939d3c72206 should also be brought back.

**Pull request merged**